### DISCUSSION TITLE
Alert suppression instructions?

### DISCUSSION THREAD

**Discussion by innovate-support** [Security-Onion-Solutions/securityonion#3257] [Unsupported Versions]
I've got a lot of known alerts from various sources.  I am successfully able to suppress **some** of them with BPF and Thresholding:sids rules in my global.sls file, but I'm not familiar with how to suppress the specific **ossec** and **windows_eventlog** alerts for specific hosts.  I don't want to completely suppress all those ossec and windows_eventlog alerts, just the ones from specific hosts.  Anyone have samples of how to accomplish this? 

SO is great, but alerts are too noisy and need an easy way of controlling them.  Not just suppression, but also a view of what custom suppression rules have been implemented and a toggle to enable/disable those suppressions, similar to the existing Acknowledge functionality.

**Comment by innovate-support** [comment]
Ideally I'd like to suppress **ossec** and **windows_eventlog** alerts based on **rule.uuid** and either **agent.ip** or **agent.name**

**Comment by amlamarra** [comment]
I'd like to chime in here and say that I, too, would like to suppress certain ossec alerts but have found the documentation on that to be lacking.

**Comment by defensivedepth** [comment]
@innovate-support @amlamarra The specifics of alert tuning depends on which alerting engine generated the alert. It sounds like you both are referring to just OSSEC/Wazuh alerts, correct?

**Reply by amlamarra** [reply]
I am. I started reading Wazuh documentation. It looks like I can raise the `log_alert_level` value in the `/opt/so/conf/wazuh/ossec.conf` file. Would I need to run `so-wazuh-restart`? That will help since I don't need to see low level logs, but I'd still like to be able to disable certain alerts.

**Reply by innovate-support** [reply]
yes, Wazuh/ossec is the engine, but they also display as 'windows_eventlog'
![Screenshot from 2021-03-09 09-57-08](https://user-images.githubusercontent.com/20424476/110508196-ff941080-80bd-11eb-9b0f-6905166813ca.png)

**Reply by defensivedepth** [reply]
For Wazuh alerts, you can raise the `log_alert_level` as a global value or override specific Wazuh rules to set a lower alert_level. We are looking at making this easier in the near future.

**Reply by innovate-support** [reply]
I changed the log_alert_level from the default 1 up to 5, then I ran `sudo so-wazuh-restart` as instructed in the SO docs, but it didn't seem to affect anything.  New non-security agent alerts are still getting flagged.  I then changed the  log_alert_level up to 12 and re-ran `sudo so-wazuh-restart` but that didn't stop any new agent alerts from flagging either.  
**_What's the proper way to get ossec.conf changes to refresh?_**

For those looking at what the ossec alert numbers 0-15 actually mean.
https://www.ossec.net/docs/manual/rules-decoders/rule-levels.html

**Reply by innovate-support** [reply]
I rebooted the Manager and both Sensors, and still Level 1 alerts are showing up.  Changing the ossec.conf setting doesn't seem to do anything.

**Reply by defensivedepth** [reply]
Be aware that existing alerts will still be in the SOC Alerts console even after changes are made - you will need to Acknowledge them.

If you are still seeing Wazuh alerts, we will need to dig deeper and see what else is going on.

**Reply by innovate-support** [reply]
Yes, I understand that previous alerts will still appear.  I'm saying that all low level alerts are still being flagged even after the conf changes are made and wazuh is restarted.  Even after a system reboot. Am I editing the wrong conf file or something?  I'm editing
`/opt/so/conf/wazuh/ossec.conf`

**Reply by innovate-support** [reply]
**_Update_**: making changes to the `/opt/so/conf/wazuh/ossec.conf` file on the **Manager** seems to have absolutely no affect on the Alert levels being flagged.  I changed the `log_alert_level` on each of my **Sensors** and then ran `sudo so-wazuh-restart` on each **Sensor** and it is now suppressing the low level alerts as desired.  

Isn't the **Manager** supposed to be controlling this?

**Reply by SanibelJack** [reply]
After adjusting the conf file would we run so-rule-update and then state.highstate to push to sensors?

**Reply by ethack** [reply]
My understanding is that files under `/opt/so/conf/` shouldn't be edited manually because Salt will overwrite them every 15 minutes or after the service restart command is run. (You should be able to tell if that's the case by checking `/opt/so/conf/wazuh/ossec.conf` after your change is made to see if it survived the Wazuh service restart.)

What I would expect to work is to copy `/opt/so/saltstack/default/salt/wazuh/files/server/ossec.conf` to `/opt/so/saltstack/local/salt/wazuh/files/server/ossec.conf` on the manager and modify your value in that file. (If this works right then `/opt/so/conf/wazuh/ossec.conf` should have your change persist after Wazuh service restart.)

Apologies if that doesn't work. I haven't done this personally; it's just based on my limited understanding of Security Onion's config.
